journal/src/main/scala/com/evolutiongaming/kafka/journal/util/ExpiringRef.scala

```diff
@@ -0,0 +1,73 @@
+package com.evolutiongaming.kafka.journal.util
+
+import cats.effect.implicits.effectResourceOps
+import cats.effect.kernel.{DeferredSource, RefSource}
+import cats.effect.syntax.all._
+import cats.effect.{Deferred, Ref, Resource, Temporal}
+import cats.syntax.all._
+import com.evolutiongaming.catshelper.Log
+import com.evolutiongaming.retry.Retry.implicits._
+import com.evolutiongaming.retry.{OnError, Retry, Strategy}
+
+import scala.concurrent.duration._
+
+trait ExpiringRef[F[_], A] extends RefSource[F, A] {
```

Comment:
I do not see how scheduled expiration will fit together with full `Ref` API, while `RefSource` provides enough possibilities. In case you have opposite opinion please share expected behaviour for `Ref#set` - should it drop expiration schedule or leave it untouched.

Reply:
I would say that (auto) reloadable ref should not provide API for manual value update at all